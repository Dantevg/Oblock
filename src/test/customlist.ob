List_ = {
	length = 0
	
	add(item) => {
		this.length := this.length + 1
		this.(this.length) := item
	}
	
	iterate() => {
		var i = 0
		self = this
		return () => {
			i += 1
			return self.(i), i
		}
	}
	
	toString() => {
		var str = "["
		for item in this do str := str ++ item ++ ", "
		return str ++ "]"
	}
}

l = List_.clone()
l.add("hey")
l.add(10)
print(l.toString())
print ["hey", 10]
