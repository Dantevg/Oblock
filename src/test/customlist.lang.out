[hey, 10, ]
[hey, 10]
